src/identity.js

```diff
@@ -417,7 +417,8 @@ export class Identity extends EventEmitter {
         const promiseFn = async (resolve, reject) => {
             const postProcess = (sessionData) => {
                 if (sessionData.error) {
-                    throw new SDKError('HasSession endpoint returned an error', sessionData.error);
+                    this.emit('error', sessionData.error);
+                    reject(new SDKError('HasSession failed', sessionData.error));
```

Comment:
The reason we can't just `throw` here is that `postProcess` is called asyncronously. This means that the `executor` is not throwing synchronously, which means that what is actually rejected is the `Promise` returned by the `executor`, rather than the `Promise` the `executor` is resolving. This could probably be found by ESLints [`no-async-promise-executor`](https://eslint.org/docs/rules/no-async-promise-executor)

Reply:
Basically, we end up in this situation as described in the rule:

> If an `async` executor function throws an error, the error will be lost and won’t cause the newly-constructed Promise to reject. This could make it difficult to debug and handle some errors.